### PR TITLE
Sketcher: initialize geometry history limit variable

### PR DESCRIPTION
**Pull request by pinkavaj**
The variable seems to be uninitialized, valgrind complais about it.

The `git grep geoHistoryLevel` and the code ispection suggest it is never set. If initialized to zero, most of the related code is never used, which can point to the fact there is either another bug or part of the code is dead.

The variable was introduced in 42bf92ad12.

The initialization to 0 ensures stable, predictable behavior, @realthunder might have a suggestion what is going on here?

### CODE REVIEW

**Comment by luzpaz** [comment]
Please fix the commit title (not the github title) from geometry instead of getometry. TIA

**Comment by chennes** [comment]
Thanks for the PR -- @bgbsww any idea what's going on here? I don't see how this can ever be non-zero (given the initialization here, anyway!).

**Comment by bgbsww** [comment]
LS3 initialization: `geoHistoryLevel = hGrpp->GetInt("GeometryHistoryLevel",1);`  It is apparently a controllable 0 or 1 option that has to do with reusing geometry IDs in sketches, but the logic is beyond me at a glance.  As you can see, the default is actually one, and we do have the same conditional checks in our code.

**Comment by pinkavaj** [comment]
Would make sense to use on the "simpler" logic (current PR) before 1.0 release and create a separate MR and enable the logic (with possibly adding the config option) after 1.0? This seems IMHO safest.

The second option is to set it to 1 by default (IMHO still worth to consider) and the third one is to backport the rest of the logic from  LS3 (I can try, but no promises).

**Comment by pinkavaj** [comment]
FYI: I compared the current code to the one in LS3 and as far as I can tell it should work identically, there were just some minor refactors, use of `auto` etc ...

**Comment by bgbsww** [comment]
I would tend towards copying the LS3 value and behavior, but would love an example of where it has an effect.

**Comment by chennes** [comment]
@pinkavaj are you continuing to work on this, or would you like it to be merged as-is?

**Comment by bgbsww** [comment]
At a minimum, shouldn't the value be 1?

**Comment by pinkavaj** [comment]
I would suggest to merge this as is (it is safest). I have another MR which imports complete SL3 behaviouir, we can discus it separately and not block this. I would go for merge as-is.

**Comment by chennes** [comment]
I'm persuaded that we should default to 1, I'm going to tweak before merging.

File: src/Mod/Sketcher/App/SketchObject.cpp [hunk 1]
```diff
@@ -156,6 +156,7 @@ SketchObject::SketchObject()
                       (App::PropertyType)(App::Prop_None),
                       "Tolerance for fitting arcs of projected external geometry");
     geoLastId = 0;
+    geoHistoryLevel = 0;
```
Comment (chennes) [inline src/Mod/Sketcher/App/SketchObject.cpp:159]:
```suggestion
    geoHistoryLevel = 1;
```

**Comment by pinkavaj** [comment]
Related RM https://github.com/FreeCAD/FreeCAD/pull/16520

**Comment by yorikvanhavre** [comment]
backported

**Pull request merged**